### PR TITLE
Create documentation for persistent user sessions

### LINKED ISSUE #29218: Write documentation for persistent sessions
This should update the regular documentation and the release notes.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #29218

Follow-up issue to update the KCB scripts: https://github.com/keycloak/keycloak-benchmark/issues/805

### CODE REVIEW

File: docs/documentation/release_notes/topics/25_0_0.adoc [hunk 1]
```diff
@@ -34,6 +34,31 @@ New options are activated by default, so {project_name} will not recognize the o
 
 For information on how to migrate, see the link:{upgradingguide_link}[{upgradingguide_name}].
 
+= Persistent user sessions
```
Comment (mhajas) [inline docs/documentation/release_notes/topics/25_0_0.adoc:37]: Would it make sense to add an information that existing online sessions (prior to the upgrade) will not survive the upgrade to 25?

File: docs/documentation/server_admin/topics/sessions/offline.adoc [hunk 2]
```diff
@@ -22,7 +22,14 @@ To issue an offline token, users must have the role mapping for the realm-level
...
+Offline sessions are besides the Infinispan caches stored also in the database. Whenever the {project_name} server is restarted or an offline session is evicted from the Infinispan cache, it is still available in the database. Any following attempt to access the offline session will load the session from the database, and also import it to the Infinispan cache.
+
+To reduce memory requirements, we introduced a configuration option to shorten lifespan for imported offline sessions. Such sessions will be evicted from the Infinispan caches after the specified lifespan, but still available in the database. This will lower memory consumption, especially for deployments with a large number of offline sessions. Currently, the offline session lifespan override is disabled by default.
+ifeval::[{project_community}==true]
+This is override is only available if the feature `persistent-user-session` is disabled.
```
Comment (mhajas) [inline docs/documentation/server_admin/topics/sessions/offline.adoc:29]:
```suggestion
This override is only available if the feature `persistent-user-session` is disabled.
```

File: docs/guides/server/caching.adoc [hunk 3]
```diff
@@ -118,6 +135,14 @@ The following caches are used to store offline sessions:
 
 Upon a cluster restart, offline sessions are lazily loaded from the database and kept in a shared cache using the two caches above.
 
+ifeval::[{project_community}==true]
+
+With this feature `persistent-user-session` enabled, the in-memory caches for offline user sessions and offline client sessions are limited to 10000 entries which will reduce the overall memory usage of Keycloak for larger installations.
```
Comment (mhajas) [inline docs/guides/server/caching.adoc:140]:
```suggestion
With feature `persistent-user-session` enabled, the in-memory caches for offline user sessions and offline client sessions are limited to 10000 entries which will reduce the overall memory usage of Keycloak for larger installations.
```

**Review by mhajas — COMMENTED** [review]
Thank you for the documentation update @ahus1! This looks mostly good to me, I just added a few comments.

**Comment by ahus1** [comment]
@mhajas - this is now ready for a final review. I also changed the feature to "preview", and aligned that with Stian in a call today.

File: docs/documentation/release_notes/topics/25_0_0.adoc (hunk 1 above)
Comment (mhajas) [reply docs/documentation/release_notes/topics/25_0_0.adoc:37]: The migration might be possible when external ISPN is used, however, it is impossible when only embedded ISPN is used which is I assume the most common setup. What I am afraid of here is that people will expect they won't lose sessions when they enable this feature for their existing setups which is not true.

Comment (ahus1) [reply docs/documentation/release_notes/topics/25_0_0.adoc:37]: OK, then please add a sentence that describes the current functionality. If we merge the other issue, we can revise it.

Comment (mhajas) [reply docs/documentation/release_notes/topics/25_0_0.adoc:37]: Added

**Review by mhajas — APPROVED** [review]
Thank you @ahus1, I added two more commits. Please check and merge if you agree with my suggestions. I am approving.

**Pull request merged**